node/cli/cli.go

```diff
@@ -0,0 +1,108 @@
+package cli
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/virtual-kubelet/virtual-kubelet/node/cli/internal/commands/providers"
+	"github.com/virtual-kubelet/virtual-kubelet/node/cli/internal/commands/root"
+	"github.com/virtual-kubelet/virtual-kubelet/node/cli/internal/commands/version"
+	"github.com/virtual-kubelet/virtual-kubelet/node/cli/provider"
+)
+
+// CommandOption sets an option on the command.
+type CommandOption func(*Command)
+
+// Command builds the CLI command
+type Command struct {
+	cmd        *cobra.Command
+	s          provider.Store
+	name       string
+	setupTrace TracingSetupFunc
+}
+
+// WithProvider registers a provider which the cli can be initialized with.
+func WithProvider(name string, f provider.InitFunc) CommandOption {
+	return func(c *Command) {
+		if c.s == nil {
+			c.s = provider.NewStore()
+		}
+		c.s.Register(name, f)
+	}
+}
+
+// WithCLIBaseName sets the name of the command.
+// This is used for things like help output.
+//
+// If not set, the name is taken from `filepath.Base(os.Args[0])`
+func WithCLIBaseName(n string) CommandOption {
+	return func(c *Command) {
+		c.name = n
+	}
+}
+
+type TracingSetupFunc func(opts.Tracing)
+
+func WithTracerSetup(f TracingSetupFunc) CommandOption {
+	return func(c *Command) {
+		c.setupTrace = f
+	}
+}
+
+// New creates a new command.
+// Call `Run()` on the returned object to run the command.
+func New(ctx context.Context, opts ...CommandOption) *Command {
+	var c Command
+	for _, o := range opts {
+		o(&c)
+	}
+
+	name := c.name
+	if name == "" {
+		name = filepath.Base(os.Args[0])
+	}
+
+	var flagOpts root.Opts
+	optsErr := root.SetDefaultOpts(&flagOpts)
```

Comment:
I still wish that we called root.NewOps instead.

Reply:
Yeah, we can do that.